src/screens/Upload.js

```diff
@@ -1,57 +1,177 @@
-import auth from '@react-native-firebase/auth';
+import React, {useState, useContext} from 'react';
+import {
+  Image,
+  PixelRatio,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  Platform,
+} from 'react-native';
+import {Button, TextInput, ToggleButton} from 'react-native-paper';
+import ImagePicker from 'react-native-image-picker';
 import {AuthContext} from '../navigation/AuthNavigator';
+import storage from '@react-native-firebase/storage';
+import firestore from '@react-native-firebase/firestore';
+import RNFetchBlob from 'rn-fetch-blob';
+import {uniqueId} from '@common/functions';
+import {useNavigation} from '@react-navigation/native';
 
-export default function UploadScreen() {
+export default function SearchScreen() {
   const user = useContext(AuthContext);
+  const [avatarSource, setAvatarSource] = useState(null);
+  const [caption, setCaption] = useState('');
+  const {navigate} = useNavigation();
 
-  async function logOut() {
-    try {
-      await auth().signOut();
-    } catch (e) {
-      console.error(e);
-    }
+  function selectPhotoTapped() {
+    const options = {
+      quality: 1.0,
+      maxWidth: 500,
+      maxHeight: 500,
+      storageOptions: {
+        skipBackup: true,
+      },
+    };
+
+    ImagePicker.showImagePicker(options, response => {
+      console.log('Response = ', response.uri);
+
+      if (response.didCancel) {
+        console.log('User cancelled photo picker');
+      } else if (response.error) {
+        console.log('ImagePicker Error: ', response.error);
+      } else if (response.customButton) {
+        console.log('User tapped custom button: ', response.customButton);
+      } else {
+        let source = null;
+
+        if (Platform.OS === 'ios') {
+          source = {uri: response.uri.replace('file:', '')};
+        } else {
+          source = {uri: response.uri};
+        }
+
+        setAvatarSource(source);
+      }
+    });
+  }
+
+  function onSubmit() {
+    const image = avatarSource.uri;
+    let imageId = uniqueId();
+    const Blob = RNFetchBlob.polyfill.Blob;
+    const fs = RNFetchBlob.fs;
+    window.XMLHttpRequest = RNFetchBlob.polyfill.XMLHttpRequest;
+    window.Blob = Blob;
+    let uploadBlob = null;
+    const imageRef = storage()
+      .ref(`Image/${user.uid}`)
+      .child(imageId + '.png');
+    let mime = 'image/png';
+
+    fs.readFile(image, 'base64')
+      .then(data => {
+        return Blob.build(data, {type: `${mime};BASE64`});
+      })
+      .then(blob => {
+        uploadBlob = blob;
+        console.log(uploadBlob._ref);
+        return imageRef.putFile(uploadBlob._ref, {contentType: mime});
+      })
+      .then(() => {
+        uploadBlob.close();
+        return imageRef.getDownloadURL();
+      })
+      .then(uri => {
+        processUpload(uri);
+      })
+      .catch(error => {
+        console.log('Error when uploading image to database', error);
+      });
+  }
+
+  function processUpload(uri) {
```

Comment:
Need error handling

Reply:
Please write pre-conditions and post-conditions so we can write unit tests